sympy/polys/matrices/sdm.py

```diff
@@ -94,6 +94,11 @@ def __rmul__(a, b):
         else:
             return NotImplemented
 
+    def scalarmul(A, lamda):
+        """ Internal method of DomainMatrix.scalarmul for Scalar multiplication"""
+        scalmul = unop_dict(A, lambda aij: aij*lamda)
+        return A.new(scalmul, A.shape, A.domain)
```

Comment:
This should check the domain.

Reply:
Also commutativity should be considered. Is this `lambda * A` or `A * lambda`?

Reply:
> This should check the domain.

Before calling `SDM.scarmul`, we already have unified the domains of `A` and `lamda`.

> Also commutativity should be considered.

I am not sure if I follow. 

Since we are multiplying every element of A with lamda, it should be `A * lambda`.

Reply:
> This should check the domain.

Before calling `SDM.scarmul`, we already have unified the domains of `A` and `lamda`.

> Also commutativity should be considered.

I am not sure if I follow. 

It is `A * lambda`.

Reply:
> Before calling `SDM.scarmul`, we already have unified the domains of `A` and `lamda`.

Oh, okay.

> Also commutativity should be considered.

At the moment there are no poly domains that are non-commutative but in principle those could be added in future. We should be careful not to assume commutativity in scalar multiplication so if the user does
`lambda * A` then the elements should be computed as `lambda * Aij`. If the user does `A * lambda` then it should be computed as `Aij * lambda`.

Reply:
> `lambda * A` then the elements should be computed as `lambda * Aij`. If the user does `A * lambda` then it should be computed as `Aij * lambda`.

If the user provides `lambda * A`, with lambda as `DomainScalar`, then also we will call the `SDM.scalarmul` with `A * lambda`, thus there won't be any issue of commutativity then.